### PR TITLE
dbt-materialize: add lag tolerance arg to dbt deploy

### LINKED ISSUE #28831: dbt-materialize: Add lag tolerance argument to dbt deploy operations
### Feature request

The `deploy_promote` and `deploy_await` operations currently lack the ability to specify a custom lag tolerance when waiting for deployment readiness.

This feature is needed to allow users more fine-grained control over the deployment process, especially in environments with varying levels of dataflow lag latency.

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

This PR introduces a new `lag_tolerance` argument to the `deploy_promote` and `deploy_await` operations in the db adapter. This allows users to specify a custom maximum acceptable lag duration when waiting for deployment readiness, providing a better control over the deployment process.

Fixes #28831

### CODE REVIEW

File: doc/user/content/manage/dbt/development-workflows.md [hunk 1]
```diff
@@ -365,17 +365,27 @@ deployment environment to ensure it's safe to [cutover](#cutover-and-cleanup).
...
 
-    The default poll interval is `15` seconds, but can be configured to a
-    different value by passing the `poll_interval=<value>` argument in the macro
-    invocation.
+    By default, `deploy_await` waits for all objects to have a **lag of less than 1 second**
+    before returning successfully. You can customize this behavior with the following arguments:
 
-1. Once `deploy_await` returns successfully, you can manually run tests against
+    - `poll_interval`: The time (in seconds) between each readiness check. Default is 15 seconds.
+    - `lag_tolerance`: The maximum acceptable lag duration. Default is '1s' (1 second).
```
Comment (morsapaes) [inline doc/user/content/manage/dbt/development-workflows.md:378]:
```suggestion
    - `lag_threshold`: The maximum lag threshold, which determines when all objects in the environment are considered hydrated and it's safe to perform the cutover step. Default: '1s' (1 second).
```

Comment (morsapaes) [reply doc/user/content/manage/dbt/development-workflows.md:378]: @bobbyiliev, let's s/`lag_tolerance`/`lag_threshold`. I'll push documentation edits in a separate commit, in the meantime.

Comment (morsapaes) [reply doc/user/content/manage/dbt/development-workflows.md]: Pushed the changed in [c8f165f846719ad937b1342b7573329f3278846](https://github.com/MaterializeInc/materialize/pull/28832/commits/c8f165f846719ad937b1342b7573329f32788468). Holler if you disagree!

Comment (bobbyiliev) [reply doc/user/content/manage/dbt/development-workflows.md]: Oh awesome! Thank you @morsapaes 🙌 I was just getting to this now.

I like the new `lag_threshold` better!

**Review by morsapaes — APPROVED** [review]
Thanks, @bobbyiliev! Can you handle cutting a new release and pinging the relevant customer channel?

**Pull request merged**